binding/go/porcupine.go

```diff
@@ -246,17 +247,65 @@ func (porcupine *Porcupine) Process(pcm []int16) (keywordIndex int, err error) {
 	return index, nil
 }
 
-func getOS() string {
+func getOS() (string, string) {
 	switch os := runtime.GOOS; os {
 	case "darwin":
-		return "mac"
+		return "mac", "x86_64"
 	case "linux":
-		return "linux"
+		osName, cpu := getLinuxDetails()
+		return osName, cpu
 	case "windows":
-		return "windows"
+		return "windows", "amd64"
 	default:
 		log.Fatalf("%s is not a supported OS", os)
-		return ""
+		return "", ""
+	}
+}
+
+func getLinuxDetails() (string, string) {
+	var archInfo = ""
+
+	if runtime.GOARCH == "amd64" {
+		return "linux", "x86_64"
+	} else if runtime.GOARCH == "arm64" {
+		archInfo = "-aarch64"
+	}
+
+	cmd := exec.Command("cat", "/proc/cpuinfo")
+	cpuInfo, err := cmd.Output()
+
+	if err != nil {
+		log.Fatalf("Failed to get CPU details: %s, defaulting to linux x86_64", err.Error())
```

Comment:
log.Fatalf kills the application as soon as possible, but runs this print first. In this case, I believe we actually do want it to fail instead of reverting to linux, so maybe just print failed to get cpu details.